File: DONGUK/프로그래머스_WEEK 8/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    people.sort()
    
    q = deque(people)
    
    while q:
        top = q[0]
        
        if top + q[-1] <= limit:
            if q:
                q.popleft()
            if q:
                q.pop()
            answer += 1
            
        else:
            if q:
                q.pop()
                answer += 1
            
    return answer

File: KIJUNG/PRO_WEEK_06/구명보트.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0, idx = 0;
    sort(people.begin(),people.end());
    while(idx<people.size()){
        if(people.back()+people[idx]<=limit){
            idx++;
        }
        people.pop_back();
        answer++;
    }
    return answer;
}